consensus/bor/heimdallgrpc/state_sync.go

```diff
@@ -0,0 +1,49 @@
+package heimdallgrpc
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/consensus/bor/clerk"
+
+	proto "github.com/maticnetwork/polyproto/heimdall"
+)
+
+func (h *HeimdallGRPCClient) StateSyncEvents(ctx context.Context, fromID uint64, to int64) ([]*clerk.EventRecordWithTime, error) {
+	eventRecords := make([]*clerk.EventRecordWithTime, 0)
+
+	req := &proto.StateSyncEventsRequest{
```

Comment:
I am more than sure that it's possible to remove all this manual unmarshal. By the time, as we discussed, let's keep it as it as. Could you create a task for bor and heimdall to find a bit better solution? We can rethink that place later.

Reply:
@0xKrishna could we update PRs (Heimdall and Bor), fix CI and merge gRPC PRs?

Reply:
Lint and CI and fixed, Working on few comments above